pkg/ssh/forward.go

```diff
@@ -11,13 +11,63 @@ import (
 	"golang.org/x/crypto/ssh"
 )
 
-func PortForward(ctx context.Context, client *ssh.Client, localNetwork, localAddr, remoteNetwork, remoteAddr string, exitAfterTimeout time.Duration, log log.Logger) error {
+type ForwardingFunction func(
+	net.Conn,
+	*ssh.Client,
+	string,
+	string,
+	log.Logger,
+)
+
+func PortForward(
+	ctx context.Context,
+	client *ssh.Client,
+	localNetwork, localAddr, remoteNetwork, remoteAddr string,
+	exitAfterTimeout time.Duration,
+	log log.Logger,
+) error {
 	listener, err := net.Listen(localNetwork, localAddr)
 	if err != nil {
 		return err
 	}
 	defer listener.Close()
 
+	return portForwarding(
+		ctx, client, listener,
+		localNetwork, localAddr, remoteNetwork, remoteAddr,
+		exitAfterTimeout, log, forward,
+	)
+}
+
+func ReversePortForward(
+	ctx context.Context,
+	client *ssh.Client,
+	remoteNetwork, remoteAddr, localNetwork, localAddr string,
+	exitAfterTimeout time.Duration,
+	log log.Logger,
+) error {
+	listener, err := client.Listen(remoteNetwork, remoteAddr)
+	if err != nil {
+		return err
+	}
+	defer listener.Close()
+
+	return portForwarding(
+		ctx, client, listener,
+		remoteNetwork, remoteAddr, localNetwork, localAddr,
+		exitAfterTimeout, log, reverseForward,
+	)
+}
+
+func portForwarding(
+	ctx context.Context,
+	client *ssh.Client,
+	listener net.Listener,
+	srcNetwork, srcAddr, dstNetwork, dstAddr string,
```

Comment:
`srcNetwork` is not used

Reply:
That's needed in order for `ForwardingFunction` alias/type to work and return the function (eithe regular or reverse forward) to be returned to `portForwarding`